src/closure_decorator.rs

```diff
@@ -321,10 +279,10 @@ impl VisitMut for ClosureDecorator {
 impl ClosureDecorator {
   fn register_stmt_if_func_decl(&mut self, stmt: &Stmt) -> Option<Stmt> {
     match stmt {
-      Stmt::Decl(Decl::Fn(func)) => Some(self.register_ast_stmt(
-        Box::new(Expr::Ident(func.ident.clone())),
-        self.parse_function_decl(&func),
-      )),
+      Stmt::Decl(Decl::Fn(func)) => {
+        let parse_func = self.parse_function_decl(&func);
+        Some(self.register_ast_stmt(Box::new(Expr::Ident(func.ident.clone())), parse_func))
```

Comment:
this change was necessary because `self.parse_function_decl` now mutably borrows `self`.